test/unit/preact/test-base-element-mapping.js

```diff
@@ -964,4 +964,47 @@ describes.realWin('PreactBaseElement', spec, (env) => {
       expect(component).to.be.calledTwice;
     });
   });
+
+  describe('delegatesFocus mapping', () => {
+    let element;
+    let style;
+
+    beforeEach(async () => {
+      Impl['delegatesFocus'] = true;
+      Impl['passthroughNonEmpty'] = true;
+      element = html`
+        <amp-preact layout="fixed" width="100" height="100"></amp-preact>
+      `;
+
+      style = html`<style>
+        amp-preact:focus {
+          outline: #0389ff solid 2px;
+        }
+      </style>`;
+      doc.body.appendChild(element);
+      doc.body.appendChild(style);
+      await element.build();
+      await waitFor(() => component.callCount > 0, 'component rendered');
+    });
+
+    it('should focus on the host when an element in the shadow DOM receives focus', async () => {
+      // initial focus is not on host
+      expect(doc.activeElement).to.not.equal(element);
+      expect(win.getComputedStyle(element)['outline']).to.equal(
+        'rgb(0, 0, 0) none 0px'
+      );
+
+      // focus an element within the shadow DOM
+      const inner = element.shadowRoot.querySelector('#component');
+      // required to receive focus
+      inner.setAttribute('tabIndex', 0);
+      inner.focus();
+
+      // host receives focus and custom style for outline
+      expect(doc.activeElement).to.equal(element);
+      expect(win.getComputedStyle(element)['outline']).to.equal(
+        'rgb(3, 137, 255) solid 2px'
+      );
```

Comment:
@jridgewell @samouri Same issue as above.  This test only passes the first time when run in `--watch` mode.  The second time this assertion will fail and expect `'rgb(0, 0, 0) none 0px'`.